---
title: "CaTS-ATAC PCA and correlation heatmaps"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# ATAC analysis


```{r message=FALSE}

rm(list=ls())

library(DESeq2)
library(RColorBrewer)
library(tidyverse)
library(ComplexHeatmap)
library(circlize)

```

## Load data

Counts table and annotation generated by nf-core/atacseq in: `cats_atac/results/bwa/mergedLibrary/macs/broadPeak/consensus/`
Further elements filtered with `cats-atac_1_filter_fdr.Rmd` 

```{r }

#counts table
count.table <- read.table(file="/Users/delasj/Documents/BriscoeLab/project_DV_ATAC_reproduce_analysis/inputs_cats-atac_2_pca_heatmap/consensus_peaks.mLb.clN.featureCounts.txt",header=TRUE, stringsAsFactors = FALSE)

# fdr per sample and interval - if peak overalapped
fdr.table <- read.table(file="/Users/delasj/Documents/BriscoeLab/project_DV_ATAC_reproduce_analysis/outputs_cats-atac_1/consensus_peakfdr_filtered.csv", header=TRUE, stringsAsFactors = FALSE)

# clean colnames
colnames(count.table) <- gsub(".mLb.clN.bam","",colnames(count.table))

# we do not need coordinates
count.table <- count.table %>%
  select("Geneid", starts_with("D"))

#keep only the filtered intervals
count.table <- count.table %>%
  filter(Geneid %in% fdr.table$interval_id)


## Annotation table
ann.table <- read.table(file="~/Documents/BriscoeLab/Project_ATAC_DV/20-08-04_ATAC_newRun_fdr_cutoff/consensus_peaks.mLb.clN.annotatePeaks.txt", header=TRUE, stringsAsFactors = FALSE, sep = "\t")
colnames(ann.table)[1] <- "Peakid"


ann.table.clean <- ann.table %>% 
  select(c("Peakid","Chr","Start","End","Strand","Annotation","Distance.to.TSS","Nearest.PromoterID")) %>%
  separate(Annotation, into = "Annotation_brief", sep = " ", remove = FALSE)


```


## Colors and shapes

```{r }
sorted.DayGate <- c("D3_NMP","D4_1","D4_2","D4_M","D4_3",
                    "D5_1","D5_2","D5_M","D5_3",
                    "D6_1","D6_2","D6_M","D6_3")
colorJD <- c("#878787","#6da4ba","#f0be56","#ec936f","#5bb357",
             "#477d92","#e5a114","#e3602b","#009640",
             "#2e525e","#9f7113","#ab4117","#044a23")
shapes4_manual = c(18,15,16,17) # these are block
shapes5_manual = c(25,21,22,23,24) # these are filled
shapes4_fill_manual = c(23,21,22,24)

# for Days
red_colors <- c("#fadede","#f3aaaa","#e96666","#cf1e1e")

# for SAG
grey_colors <- c("#f6f6f6","#cecece","#808080","#595959")

```

## Differential analysis

```{r}

count.matrix <- count.table %>%
  column_to_rownames("Geneid")

## Make metadata file for DESeq

genecolData_first <- data.frame(Sample_ID = colnames(count.matrix))
genecolData_first <- genecolData_first %>% 
  separate(Sample_ID,into=c("Day","SAG","Gate","Rep"), sep="_", remove=FALSE) %>%
  mutate(Condition=paste(Day,SAG,Gate, sep="_"),
         DaySAG=paste(Day,SAG,sep = "_"),
         DayGate=paste(Day,Gate,sep="_"),
         Experiment=paste(SAG,Rep,sep="_"))
genecolData_first <- as.data.frame(unclass(genecolData_first))


dds <- DESeqDataSetFromMatrix(countData = count.matrix,
                              colData = genecolData_first,
                              design = ~ Gate)

dds <- DESeq(dds)

vsd <- varianceStabilizingTransformation(dds,blind = FALSE)

```

## Export files 
Useful for ploting heatmaps elsewhere

```{r }
# Export normalized tables for plotting elsewhere
dds_counts <- counts(dds, normalized = TRUE)
vsd_data <- assay(vsd)

write.table(dds_counts, file = "/Users/delasj/Documents/BriscoeLab/project_DV_ATAC_reproduce_analysis/outputs_cats-atac_2/consensus_peaks.mLb.clN.normCounts.txt", quote = FALSE, row.names = TRUE)
write.csv(vsd_data, file = "/Users/delasj/Documents/BriscoeLab/project_DV_ATAC_reproduce_analysis/outputs_cats-atac_2/consensus_peaks.mLb.vsd.csv", quote = FALSE)

```

## Plot PCAs 

```{r fig.width=5, fig.height=4}
# calculate the variance for each gene
rv <- rowVars(vsd_data)
# select the ntop genes by variance
ntop=30000
select <- order(rv, decreasing=TRUE)[seq_len(min(ntop, length(rv)))]

t_vsd <- t(vsd_data[select,])
vsd_pca <- prcomp(t_vsd, retx=TRUE, scale. = FALSE)
#summary(vsd_pca)

#names(vsd_pca)

# How many PC to explain enough variance?
#summary(vsd_pca)

var_explained <-vsd_pca$sdev^2/sum(vsd_pca$sdev^2)
#plot(var_explained)

vsd_pca_plot <- vsd_pca$x %>% 
  as.data.frame %>%
  rownames_to_column("Sample") %>%
  separate(Sample,into=c("Day","SAG","Gate","Rep"), sep="_", remove=FALSE) %>%
  mutate(Condition=paste(Day,SAG,Gate, sep="_"),
         DaySAG=paste(Day,SAG,sep = "_"),
         DayGate=paste(Day,Gate,sep="_"),
         Experiment=paste(SAG,Rep,sep="_"),
         DayGate = factor(DayGate, levels = sorted.DayGate)) 


ggplot(vsd_pca_plot, aes(x=-PC1,y=PC2,fill=Day)) +
  scale_fill_manual(values = red_colors) +
  geom_point(size=4, alpha=0.9, shape=21) +
  guides(fill = guide_legend(override.aes=list(shape=21))) +
  #scale_shape_manual(values = shapes4_fill_manual) +
  labs(x=paste0("PC1: ",round(var_explained[1]*100,1),"%"),
       y=paste0("PC2: ",round(var_explained[2]*100,1),"%")) +
  theme_bw(base_size=12)
#c(23,21,22,24)

ggplot(vsd_pca_plot, aes(x=-PC1,y=PC2,fill=SAG,shape=SAG)) +
  #scale_fill_manual(values = grey_colors) +
  scale_fill_grey(start = 0.9, end = 0.1) + 
  geom_point(size=4, alpha=0.9) +
  guides(fill = guide_legend(override.aes=list(shape=21))) +
  scale_shape_manual(values = shapes4_fill_manual) +
  labs(x=paste0("PC1: ",round(var_explained[1]*100,1),"%"),
       y=paste0("PC2: ",round(var_explained[2]*100,1),"%")) +
  theme_bw(base_size=12)

ggplot(vsd_pca_plot, aes(x=-PC1,y=PC2,fill=DayGate,shape=SAG)) +
  scale_fill_manual(values = colorJD) +
  geom_point(size=4, alpha=0.9) +
  guides(fill = guide_legend(override.aes=list(shape=21))) +
  scale_shape_manual(values = shapes4_fill_manual) +
  labs(x=paste0("PC1: ",round(var_explained[1]*100,1),"%"),
       y=paste0("PC2: ",round(var_explained[2]*100,1),"%")) +
  theme_bw(base_size=12)


```


# Correlation Heatmaps - by genomic annotation 

Merge annotation and vsd
```{r}
ann_vsd <- as.data.frame(vsd_data) %>%
  rownames_to_column("Peakid") %>%
  left_join(ann.table.clean, by="Peakid")
```


### Annotated heatmap of TSS

```{r fig.width=11,fig.height=11}

ann_vsd_TSS <- ann_vsd %>%
  filter(Annotation_brief=="promoter-TSS")

ann_vsd_TSS_hm <- ann_vsd_TSS %>%
  remove_rownames() %>%
  column_to_rownames("Peakid") %>%
  select("D5_100_M_R3":"D6_500_3_R1")

cor_vsd_TSS <-cor(ann_vsd_TSS_hm)


```


### Annotated heatmap of all but TSS
Or exon because it often includes the TSS
```{r fig.width=11,fig.height=11}

ann_vsd_nonTSS <- ann_vsd %>%
  filter(Annotation_brief %in% c("intron","Intergenic","TTS"))

ann_vsd_nonTSS_hm <- ann_vsd_nonTSS %>%
  remove_rownames() %>%
  column_to_rownames("Peakid") %>%
  select("D5_100_M_R3":"D6_500_3_R1")

cor_vsd_nonTSS <-cor(ann_vsd_nonTSS_hm)

```


## Plots heatmaps with the same scale

```{r fig.height=5.76, fig.width=6.28}


phen_data <- genecolData_first %>%
  select(c("Sample_ID","DayGate","Day","SAG")) %>%
  remove_rownames() %>%
  column_to_rownames("Sample_ID")

#colors for heatmap
ann_color_JD <- list(
  DayGate = c(D3_NMP="#878787",D4_1="#6da4ba",D5_1="#477d92", D6_1="#2e525e",
              D4_2="#f0be56",D5_2="#e5a114",D6_2="#9f7113",
              D4_M="#ec936f",D5_M="#e3602b",D6_M="#ab4117",
              D4_3="#5bb357",D5_3="#009640",D6_3="#044a23"),
  SAG = c(`0`="#f6f6f6",`10`="#cecece",`100`="#808080",`500`="#595959"),
  Day = c(D3="#fadede",D4="#f3aaaa",D5="#e96666",D6="#cf1e1e"),
  Rep = c(R1="#ebeb77", R2="#77b1eb", R3="#eb7777"))


# Build the annotation for the complex heatmap
heatmap_ann_row <- rowAnnotation(df=phen_data, col=ann_color_JD,
       annotation_legend_param = list(DayGate = list(at =c("D3_NMP","D4_1","D5_1","D6_1",
               "D4_2","D5_2","D6_2",
              "D4_M","D5_M","D6_M",
              "D4_3","D5_3","D6_3"))))

heatmap_ann <- HeatmapAnnotation(df=phen_data, 
    col=ann_color_JD, show_legend = FALSE,
    annotation_height = unit(1, 'cm'),
    annotation_width = unit(1, 'cm'),
    gap = unit(1, 'mm'))

heatmap_ann_row <- rowAnnotation(df=phen_data, col=ann_color_JD,
      annotation_height = unit(1, 'cm'),
      annotation_width = unit(1, 'cm'),
      gap = unit(1, 'mm'),
       annotation_legend_param = list(DayGate = list(at =c("D3_NMP","D4_1","D5_1","D6_1",
               "D4_2","D5_2","D6_2",
              "D4_M","D5_M","D6_M",
              "D4_3","D5_3","D6_3"))))


# Annotated heatmap with selected colors
hm_colors = colorRampPalette(rev(brewer.pal(n = 11, name = "RdBu")))(100)

col_fun = colorRamp2(seq(0.2,1,length.out= 11) , 
                     c(colorRampPalette(rev(brewer.pal(n = 11, name = "RdBu")))(11)))


hmap_notgenes<- Heatmap(cor_vsd_nonTSS, name="Correlation", col=col_fun,
        cluster_columns = TRUE, cluster_rows = TRUE,
        #show_column_dend = FALSE,
        #row_dend_side = "right", column_dend_side = "bottom",
        clustering_distance_columns = function(x) as.dist(1 - cor(t(x))),
        clustering_method_columns = 'ward.D2',
        clustering_distance_rows = function(x) as.dist(1 - cor(t(x))),
        clustering_method_rows = 'ward.D2',
        column_dend_height = unit(2, "cm"), row_dend_width = unit(2, "cm"),
        row_names_gp = gpar(fontsize = 10),column_names_gp = gpar(fontsize = 10),
        left_annotation = heatmap_ann_row, top_annotation = heatmap_ann,
        show_column_names = FALSE,show_row_names = FALSE,
        column_title = "Pearson Correlation - NonTSS or exon")

draw(hmap_notgenes, heatmap_legend_side = 'left',
        annotation_legend_side = 'right',
        row_sub_title_side = 'left')



hmap_TSS <- Heatmap(cor_vsd_TSS, name="Correlation", col=col_fun,
        cluster_columns = TRUE, cluster_rows = TRUE,
        #show_column_dend = FALSE,
        #row_dend_side = "right", column_dend_side = "bottom",
        clustering_distance_columns = function(x) as.dist(1 - cor(t(x))),
        clustering_method_columns = 'ward.D2',
        clustering_distance_rows = function(x) as.dist(1 - cor(t(x))),
        clustering_method_rows = 'ward.D2',
        column_dend_height = unit(2, "cm"), row_dend_width = unit(2, "cm"),
        row_names_gp = gpar(fontsize = 10),column_names_gp = gpar(fontsize = 10),
        left_annotation = heatmap_ann_row, top_annotation = heatmap_ann,
        show_column_names = FALSE,show_row_names = FALSE,
        column_title = "Pearson Correlation - TSS")

draw(hmap_TSS, heatmap_legend_side = 'left',
        annotation_legend_side = 'right',
        row_sub_title_side = 'left')

```




```{r}
sessionInfo()
```


