---
title: "ATACseq upon Foxa2 induction"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# ATAC analysis of inducible Foxa2 vs control

```{r message=FALSE}

rm(list=ls())

library(DESeq2)
library(RColorBrewer)
library(tidyverse)

```

### Set dirs
```{r}

workingdir="/Users/delasj/Documents/BriscoeLab/project_DV_ATAC_reproduce_analysis/"
subworkinput="inputs_ifoxa2-atac_1/"
outdir="outputs_ifoxa2-atac_1/"

ifelse(!dir.exists(file.path(workingdir,outdir)), dir.create(file.path(workingdir,outdir)), "Directory exists")

```

## Load data

Counts table generated by nf-core/atacseq in: `results/bwa/mergedLibrary/macs/broadPeak/consensus/`

```{r }

#counts table
count_table_full <- read.table(file=paste0(workingdir,subworkinput,"consensus_peaks.mLb.clN.featureCounts.txt"),header=TRUE, stringsAsFactors = FALSE)


# clean colnames
colnames(count_table_full) <- gsub(".mLb.clN.bam","",colnames(count_table_full))


# we do not need coordinates
count_table <- count_table_full %>%
  select("Geneid", starts_with("mCh"))

colnames(count_table) <- gsub("mCh.","",colnames(count_table))


```


## Colors and shapes

```{r }

Foxa2colors <- c("#a6a6a6","#82008e","#5e0067")
shapes5_manual = c(25,21,22,23,24) # these are filled
shapes4_fill_manual = c(23,21,22,24)

```

## Differential analysis

```{r}

count_matrix <- count_table %>%
  column_to_rownames("Geneid")

## Make metadata file for DESeq

genecolData_first <- data.frame(Sample_ID = colnames(count_matrix))
genecolData_first <- genecolData_first %>% 
  separate(Sample_ID,into=c("Condition","Rep"), sep="_", remove=FALSE)
genecolData_first <- as.data.frame(unclass(genecolData_first))


dds <- DESeqDataSetFromMatrix(countData = count_matrix,
                              colData = genecolData_first,
                              design = ~ Condition)

dds <- DESeq(dds)

vsd <- varianceStabilizingTransformation(dds,blind = FALSE)

```

## Export files 

Export normalized tables for plotting elsewhere

```{r }

dds_counts <- counts(dds, normalized = TRUE)
vsd_data <- assay(vsd)
dds_results <- results(dds)

# write.table(dds_counts, file = "1_output_tables/consensus_peaks.mLb.clN.normCounts.txt", quote = FALSE, row.names = TRUE)
# write.csv(vsd_data, file = "1_output_tables/consensus_peaks.mLb.vsd.csv", quote = FALSE)

```

## Plot PCAs 

This should reveal any major batch effects. It will not too informative otherwise.

```{r fig.width=4.5,fig.height=3}

# calculate the variance for each gene
rv <- rowVars(vsd_data)
# select the ntop genes by variance
ntop=30000
select <- order(rv, decreasing=TRUE)[seq_len(min(ntop, length(rv)))]


t_vsd <- t(vsd_data[select,])
vsd_pca <- prcomp(t_vsd, retx=TRUE, scale. = FALSE)


var_explained <-vsd_pca$sdev^2/sum(vsd_pca$sdev^2)

vsd_pca_plot <- vsd_pca$x %>% 
  as.data.frame %>%
  rownames_to_column("Sample") %>%
  separate(Sample,into=c("Condition","Rep"), sep="_", remove=FALSE)


ggplot(vsd_pca_plot, aes(x=PC1,y=PC2,fill=Condition,shape=Rep)) +
  scale_fill_manual(values = Foxa2colors) +
  geom_point(size=4, alpha=0.9) +
  guides(fill = guide_legend(override.aes=list(shape=21))) +
  scale_shape_manual(values = shapes4_fill_manual) +
  labs(x=paste0("PC1: ",round(var_explained[1]*100,1),"%"),
       y=paste0("PC2: ",round(var_explained[2]*100,1),"%")) +
  theme_bw(base_size=16)

```

## MA plot

Plot differentially accessible elements vs baseMean

```{r}
# color significant
results_sub_plot1 <- dds_results %>%
  as.data.frame() %>%
  mutate(color_sig=case_when(padj < 0.1 & padj > 0.01 ~ "under01",
                      padj < 0.01 & padj > 0.001 ~ "under001",
                      padj < 0.001 & padj >0 ~ "under0001",
                      TRUE ~ "over01"))
      
threshold <- 7
table(results_sub_plot1$color_sig)
      
n_diff_1 <- results_sub_plot1 %>%
        mutate(Sig0001 = case_when(padj < 0.001 & log2FoldChange > 0 ~ "up",
                                   padj < 0.001 & log2FoldChange < 0 ~ "down"))
table(n_diff_1$Sig0001)

ggplot(results_sub_plot1, aes(x=baseMean, y=log2FoldChange, color=color_sig)) +
          geom_point(size=1) +
          geom_point(data=results_sub_plot1[results_sub_plot1$log2FoldChange > threshold,],
                   aes(x=baseMean, y=threshold), shape = 2, colour="#d83a00") +
          # geom_point(data=results_sub_plot1[results_sub_plot1$log2FoldChange < -threshold,],
          #          aes(x=baseMean, y=-threshold), shape = 2, colour="#d83a00") +
          ylim(-threshold,threshold) +
          scale_x_log10() +
          scale_color_manual(values = c("gray30","#d83a00","#ff9b76","#ffd4c4")) +
          ylab(paste0("log2 Fold Change Foxa2 vs Ctrl ATAC")) +
          ggtitle(paste0("Results Foxa2 vs Ctrl")) +
          theme_bw() 


library(ggrastr)
# ggplot(results_sub_plot1, aes(x=baseMean, y=log2FoldChange, color=color_sig)) +
#           rasterise(geom_poin t(size=1),dpi = 300) +
#           geom_point(data=results_sub_plot1[results_sub_plot1$log2FoldChange > threshold,],
#                    aes(x=baseMean, y=threshold), shape = 2, colour="#d83a00") +
#           # geom_point(data=results_sub_plot1[results_sub_plot1$log2FoldChange < -threshold,],
#           #          aes(x=baseMean, y=-threshold), shape = 2, colour="#d83a00") +
#           ylim(-threshold,threshold) +
#           scale_x_log10() +
#           scale_color_manual(values = c("gray30","#d83a00","#ff9b76","#ffd4c4")) +
#           ylab(paste0("log2 Fold Change Foxa2 vs Ctrl ATAC")) +
#           ggtitle(paste0("Results Foxa2 vs Ctrl")) +
#           theme_bw() 



```


### Export diff acces peaks to overlap with WT clusters

Export bed files to perform BED intersect. Export deseq results with annotation to explore. 

```{r}

Foxa2_up <- dds_results %>% 
  as.data.frame() %>%
  filter(padj < 0.001 & log2FoldChange > 0) %>%
  rownames_to_column("Geneid")

Foxa2_down <- dds_results %>% 
  as.data.frame() %>%
  filter(padj < 0.001 & log2FoldChange < 0) %>%
  rownames_to_column("Geneid")


Foxa2_up_bed <- count_table_full %>%
  select(Chr,Start,End, Strand,Geneid) %>%
  filter(Geneid %in% Foxa2_up$Geneid)

Foxa2_down_bed <- count_table_full %>%
  select(Chr,Start,End, Strand,Geneid) %>%
  filter(Geneid %in% Foxa2_down$Geneid)

# write.csv(Foxa2_up, file="1_outputs_DESeq2results/Foxa2_up_results.csv", row.names = F, quote = F)
# write.csv(Foxa2_down, file="1_outputs_DESeq2results/Foxa2_down_results.csv", row.names = F, quote = F)
# 
# 
write.table(Foxa2_up_bed, file=paste0(workingdir,outdir,"Foxa2_up.bed"),sep="\t", row.names = F, col.names = F, quote = F)
write.table(Foxa2_down_bed, file=paste0(workingdir,outdir,"Foxa2_down.bed"),sep="\t", row.names = F, col.names = F, quote = F)

```





```{r}
sessionInfo()
```


