RNA-seq analysis
================

This is RNA-seq from PFA-fixed, permeabilized and sorted cells. Data
quality is limited.

``` r
rm(list=ls())

library(tidyverse)
library(DESeq2)
library(ggplot2)
library(RColorBrewer)
```

### Set dirs

``` r
workingdir="/Users/delasj/Documents/BriscoeLab/project_DV_ATAC_reproduce_analysis/"
subworkinput="inputs_rna_1/"
outdir="outputs_rna_1/"

ifelse(!dir.exists(file.path(workingdir,outdir)), dir.create(file.path(workingdir,outdir)), "Directory exists")
```

    ## [1] TRUE

## Load data

Data generated by `R_gene_counts`

``` r
count.table <- read.table(file=paste0(workingdir,subworkinput,"Counts_DupR.featureCounts_custom.txt"),header=TRUE, stringsAsFactors = FALSE)

#clean header
colnames(count.table) <- gsub(".read1Aligned.sortedByCoord.out.markDups.bam","", colnames(count.table))
```

## Color and order

``` r
#order conditions
sorted.gate <- c("NMP","1","2","M","3")

sorted.DayGate <- c("D3_NMP","D4_1","D4_2","D4_M","D4_3",
                    "D5_1","D5_2","D5_M","D5_3",
                    "D6_1","D6_2","D6_M","D6_3")
colorJD <- c("#878787","#6da4ba","#f0be56","#ec936f","#5bb357",
             "#477d92","#e5a114","#e3602b","#009640",
             "#2e525e","#9f7113","#ab4117","#044a23")
shapes4_manual = c(18,15,16,17) # these are block
shapes5_manual = c(25,21,22,23,24) # these are filled
shapes4_fill_manual = c(23,21,22,24)
```

## Prep matrix with just genes

``` r
count.matrix <- count.table %>%
  select(c("GeneID",starts_with("D"))) %>%
  filter(!grepl("ERCC", GeneID)) %>%
  remove_rownames %>%
  column_to_rownames("GeneID")
```

## Differential analysis

``` r
## Make metadata file for DESeq
genecolData_first <- data.frame(Sample_ID = colnames(count.matrix))
genecolData_first <- genecolData_first %>% 
  separate(Sample_ID,into=c("Day","SAG","Gate","Rep"), sep="\\.", remove=FALSE) %>%
  mutate(Condition=paste(Day,SAG,Gate, sep="_"),
         DaySAG=paste(Day,SAG,sep = "_"),
         DayGate=paste(Day,Gate,sep="_"),
         Experiment=paste(SAG,Rep,sep="_")) 

genecolData_first <- as.data.frame(unclass(genecolData_first))




dds <- DESeqDataSetFromMatrix(countData = count.matrix,
                              colData = genecolData_first,
                              design = ~ Gate)

dds <- DESeq(dds)
```

    ## estimating size factors

    ## estimating dispersions

    ## gene-wise dispersion estimates

    ## mean-dispersion relationship

    ## final dispersion estimates

    ## fitting model and testing

    ## -- replacing outliers and refitting for 2336 genes
    ## -- DESeq argument 'minReplicatesForReplace' = 7 
    ## -- original counts are preserved in counts(dds)

    ## estimating dispersions

    ## fitting model and testing

``` r
vsd <- varianceStabilizingTransformation(dds,blind = FALSE)
```

Get tables and count normalized. Export data to use elsewhere.

``` r
vsd_data <- assay(vsd)
counts_data <- counts(dds, normalized = TRUE)
```

``` r
write.csv(counts_data, file=paste0(workingdir,outdir,"RNA_normCounts_filter1.csv"), quote=FALSE)
write.csv(vsd_data, file=paste0(workingdir,outdir,"RNA_vsd_filter1.csv"), quote=FALSE)
```

## Plot marker genes

Validate the sorted populations expressed the expected genes. Marker
genes from Delile et al.Â 

``` r
GOIs <- c("T","Sp8","Pax6","Nkx6-1","Foxn4","Olig2","Nkx2-2","Nkx2-9")

counts_data_sub <- count.matrix[GOIs,]

counts_data_sub_plot <- counts_data_sub %>%
  as.data.frame() %>%
  rownames_to_column("geneid") %>%
  gather(Sample_ID,Counts_norm, starts_with("D")) %>%
  separate(Sample_ID,into=c("Day","SAG","Gate","Rep"), sep="\\.", remove=FALSE) %>%
  mutate(Condition=paste(Day,SAG,Gate, sep="_"),
         DaySAG=paste(Day,SAG,sep = "_"),
         DayGate=paste(Day,Gate,sep="_"),
         Experiment=paste(SAG,Rep,sep="_")) %>%
  mutate(DayGate = factor(DayGate, levels = sorted.DayGate),
         geneid=factor(geneid, levels = GOIs))

ggplot(counts_data_sub_plot, aes(x=SAG,y=Counts_norm,fill=DayGate)) +
    stat_summary(aes(fill=DayGate),
     fun = mean, geom="bar", alpha=0.8,position=position_dodge2(width=0.9,preserve = "single")) +
    geom_point(aes(fill=DayGate), alpha=0.7, position = position_dodge2(width=0.9,preserve ="single"),color="black",shape=21) +
  scale_fill_manual(values = colorJD) +
  facet_grid(geneid ~ Day, scales = "free", space = "free_x") +
  theme_bw()
```

![](rna_1_files/figure-gfm/unnamed-chunk-9-1.png)<!-- -->

``` r
sessionInfo()
```

    ## R version 3.6.3 (2020-02-29)
    ## Platform: x86_64-apple-darwin15.6.0 (64-bit)
    ## Running under: macOS Catalina 10.15.7
    ## 
    ## Matrix products: default
    ## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
    ## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
    ## 
    ## locale:
    ## [1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8
    ## 
    ## attached base packages:
    ## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
    ## [8] methods   base     
    ## 
    ## other attached packages:
    ##  [1] RColorBrewer_1.1-3          DESeq2_1.26.0              
    ##  [3] SummarizedExperiment_1.16.1 DelayedArray_0.12.3        
    ##  [5] BiocParallel_1.20.1         matrixStats_0.61.0         
    ##  [7] Biobase_2.46.0              GenomicRanges_1.38.0       
    ##  [9] GenomeInfoDb_1.22.1         IRanges_2.20.2             
    ## [11] S4Vectors_0.24.4            BiocGenerics_0.32.0        
    ## [13] forcats_0.5.1               stringr_1.4.0              
    ## [15] dplyr_1.0.8                 purrr_0.3.4                
    ## [17] readr_2.1.2                 tidyr_1.2.0                
    ## [19] tibble_3.1.6                ggplot2_3.3.5              
    ## [21] tidyverse_1.3.1            
    ## 
    ## loaded via a namespace (and not attached):
    ##  [1] colorspace_2.0-3       ellipsis_0.3.2         htmlTable_2.4.0       
    ##  [4] XVector_0.26.0         base64enc_0.1-3        fs_1.5.2              
    ##  [7] rstudioapi_0.13        farver_2.1.0           bit64_4.0.5           
    ## [10] AnnotationDbi_1.48.0   fansi_1.0.3            lubridate_1.8.0       
    ## [13] xml2_1.3.3             splines_3.6.3          cachem_1.0.6          
    ## [16] geneplotter_1.64.0     knitr_1.38             Formula_1.2-4         
    ## [19] jsonlite_1.8.0         broom_0.7.12           annotate_1.64.0       
    ## [22] cluster_2.1.2          dbplyr_2.1.1           png_0.1-7             
    ## [25] compiler_3.6.3         httr_1.4.2             backports_1.4.1       
    ## [28] assertthat_0.2.1       Matrix_1.3-2           fastmap_1.1.0         
    ## [31] cli_3.2.0              htmltools_0.5.2        tools_3.6.3           
    ## [34] gtable_0.3.0           glue_1.6.2             GenomeInfoDbData_1.2.2
    ## [37] Rcpp_1.0.8.3           cellranger_1.1.0       vctrs_0.4.0           
    ## [40] xfun_0.30              rvest_1.0.2            lifecycle_1.0.1       
    ## [43] XML_3.99-0.3           zlibbioc_1.32.0        scales_1.1.1          
    ## [46] hms_1.1.1              yaml_2.3.5             memoise_2.0.1         
    ## [49] gridExtra_2.3          rpart_4.1.16           latticeExtra_0.6-29   
    ## [52] stringi_1.7.6          RSQLite_2.2.12         highr_0.9             
    ## [55] genefilter_1.68.0      checkmate_2.0.0        rlang_1.0.2           
    ## [58] pkgconfig_2.0.3        bitops_1.0-7           evaluate_0.15         
    ## [61] lattice_0.20-45        labeling_0.4.2         htmlwidgets_1.5.4     
    ## [64] bit_4.0.4              tidyselect_1.1.2       magrittr_2.0.3        
    ## [67] R6_2.5.1               generics_0.1.2         Hmisc_4.6-0           
    ## [70] DBI_1.1.2              pillar_1.7.0           haven_2.4.3           
    ## [73] foreign_0.8-76         withr_2.5.0            survival_3.3-1        
    ## [76] RCurl_1.98-1.6         nnet_7.3-17            modelr_0.1.8          
    ## [79] crayon_1.5.1           utf8_1.2.2             tzdb_0.3.0            
    ## [82] rmarkdown_2.13         jpeg_0.1-9             locfit_1.5-9.4        
    ## [85] grid_3.6.3             readxl_1.4.0           data.table_1.14.2     
    ## [88] blob_1.2.2             reprex_2.0.1           digest_0.6.29         
    ## [91] xtable_1.8-4           munsell_0.5.0
