---
title: "RNA-seq analysis"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This is RNA-seq from PFA-fixed, permeabilized and sorted cells. Data quality is limited.


```{r message=FALSE}

rm(list=ls())

library(tidyverse)
library(DESeq2)
library(ggplot2)
library(RColorBrewer)

```

## Load data

Data generated by `R_gene_counts` 

```{r }

count.table <- read.table(file="/Users/delasj/Documents/BriscoeLab/project_DV_ATAC_reproduce_analysis/inputs_rna_1/Counts_DupR.featureCounts_custom.txt",header=TRUE, stringsAsFactors = FALSE)

#clean header
colnames(count.table) <- gsub(".read1Aligned.sortedByCoord.out.markDups.bam","", colnames(count.table))

```


## Color and order
```{r}

#order conditions
sorted.gate <- c("NMP","1","2","M","3")

sorted.DayGate <- c("D3_NMP","D4_1","D4_2","D4_M","D4_3",
                    "D5_1","D5_2","D5_M","D5_3",
                    "D6_1","D6_2","D6_M","D6_3")
colorJD <- c("#878787","#6da4ba","#f0be56","#ec936f","#5bb357",
             "#477d92","#e5a114","#e3602b","#009640",
             "#2e525e","#9f7113","#ab4117","#044a23")
shapes4_manual = c(18,15,16,17) # these are block
shapes5_manual = c(25,21,22,23,24) # these are filled
shapes4_fill_manual = c(23,21,22,24)




```

## Prep matrix with just genes

```{r}

count.matrix <- count.table %>%
  select(c("GeneID",starts_with("D"))) %>%
  filter(!grepl("ERCC", GeneID)) %>%
  remove_rownames %>%
  column_to_rownames("GeneID")


```
## Differential analysis

```{r}

## Make metadata file for DESeq
genecolData_first <- data.frame(Sample_ID = colnames(count.matrix))
genecolData_first <- genecolData_first %>% 
  separate(Sample_ID,into=c("Day","SAG","Gate","Rep"), sep="\\.", remove=FALSE) %>%
  mutate(Condition=paste(Day,SAG,Gate, sep="_"),
         DaySAG=paste(Day,SAG,sep = "_"),
         DayGate=paste(Day,Gate,sep="_"),
         Experiment=paste(SAG,Rep,sep="_")) 

genecolData_first <- as.data.frame(unclass(genecolData_first))




dds <- DESeqDataSetFromMatrix(countData = count.matrix,
                              colData = genecolData_first,
                              design = ~ Gate)

dds <- DESeq(dds)

vsd <- varianceStabilizingTransformation(dds,blind = FALSE)

```

Get tables and count normalized.
Export data to use elsewhere.

```{r}
vsd_data <- assay(vsd)
counts_data <- counts(dds, normalized = TRUE)

```



```{r}
write.csv(counts_data, file="/Users/delasj/Documents/BriscoeLab/project_DV_ATAC_reproduce_analysis/outputs_rna_1/RNA_normCounts_filter1.csv", quote=FALSE)
write.csv(vsd_data, file="/Users/delasj/Documents/BriscoeLab/project_DV_ATAC_reproduce_analysis/outputs_rna_1/RNA_vsd_filter1.csv", quote=FALSE)
```

## Plot marker genes
Validate the sorted populations expressed the expected genes. Marker genes from Delile et al. 

```{r fig.width=7, fig.height=6}
GOIs <- c("T","Sp8","Pax6","Nkx6-1","Foxn4","Olig2","Nkx2-2","Nkx2-9")

counts_data_sub <- count.matrix[GOIs,]

counts_data_sub_plot <- counts_data_sub %>%
  as.data.frame() %>%
  rownames_to_column("geneid") %>%
  gather(Sample_ID,Counts_norm, starts_with("D")) %>%
  separate(Sample_ID,into=c("Day","SAG","Gate","Rep"), sep="\\.", remove=FALSE) %>%
  mutate(Condition=paste(Day,SAG,Gate, sep="_"),
         DaySAG=paste(Day,SAG,sep = "_"),
         DayGate=paste(Day,Gate,sep="_"),
         Experiment=paste(SAG,Rep,sep="_")) %>%
  mutate(DayGate = factor(DayGate, levels = sorted.DayGate),
         geneid=factor(geneid, levels = GOIs))

ggplot(counts_data_sub_plot, aes(x=SAG,y=Counts_norm,fill=DayGate)) +
    stat_summary(aes(fill=DayGate),
     fun = mean, geom="bar", alpha=0.8,position=position_dodge2(width=0.9,preserve = "single")) +
    geom_point(aes(fill=DayGate), alpha=0.7, position = position_dodge2(width=0.9,preserve ="single"),color="black",shape=21) +
  scale_fill_manual(values = colorJD) +
  facet_grid(geneid ~ Day, scales = "free", space = "free_x") +
  theme_bw()
```


```{r}
sessionInfo()
```


